### ISSUE TITLE
Add Gated Linear Unit activation function

### ISSUE THREAD

**Issue by gante** [keras-team/keras#16131]
**Describe the feature and the current behavior/state**.

Hi Keras team 👋  At Hugging Face, we use the [Gated Linear Unit](https://arxiv.org/abs/1612.08083) in a few speech models. Speech is one of our growing modalities, where we have a particular focus on low-resource languages. I'm opening this issue to ask if you are interested in pushing this activation function upstream (where I'd be happy to contribute).

Our TF implementation is [here](https://github.com/huggingface/transformers/blob/master/src/transformers/activations_tf.py#L86). It takes a certain input `x`, splits one of its dimensions in two, creating two tensors (`a` and `b`), and then returns `a * sigmoid(b)`.

**Will this change the current api? How?**

- Adds a new activation function;
- (This new activation function, due to its own operation, requires the `axis` across which the input is split to have an even length, and outputs a tensor whose shape is not the same as the input tensor. This does not break the API _per se_, but it would be an activation function with a different shape-related behavior.)

**Who will benefit from this feature?**

- Speech ML researchers and practitioners. The original paper did not share the code for the activation function (AFAIK).

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes :) HF ❤️  Keras
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): See [our TF implementation](https://github.com/huggingface/transformers/blob/master/src/transformers/activations_tf.py#L86). For Keras, it would need more detailed documentation (it would land [here](https://keras.io/api/layers/activations/)), as well as tests.

**Comment by chenmoneygithub** [comment]
Hi gante, a PR is more than welcome! Thanks!
